docs/TrafficSplitting.md

```diff
@@ -0,0 +1,120 @@
+# Traffic Splitting
+
+The purpose of this section of the kn documentation is to describe the command verbs for traffic splitting operations and present some examples.
+
+`kn` client can work with following properties of a traffic target defined in traffic block of a service spec.
+ - **RevisionName**: RevisionName of a specific revision to which to send a portion of traffic.
+ - **Tag**: Tag is optionally used to expose a dedicated url for referencing this target exclusively.
+ - **Percent**: Percent specifies percent of the traffic to this Revision.
+
+## Operations
+
+Operations required to perform traffic splitting use cases, for example: blue/green deployment, etc can be summarized as below:
+ - Tag targets
+ - Split traffic
+
+### Tag Targets
+
+User can tag traffic targets (revisions) to
+ - Create custom URLs
+ - Create a human-friendly reference to refer to a traffic target
+ - Create static routes and change the underlying revisions as needed
+ - Remove the custom URLs OR remove the tags of targets
+ - Remove a target from traffic block altogether
+
+
+#### Tagging commands
+
+#### 1. `kn service tag-target [RevisionName1:Tag1] [RevisionName2:Tag2]`
+
+ - Tags given revision with specified tag thereby creating a custom URL
+ - Does not change the traffic percent of the target
+ - Multiple `RevisionName:Tag` combination can be specified
+ - Multiple tags for same revision can be specified
+   - `kn service tag-target echo-v1:stable echo-v1:current`
+ - If the target doesn't exist in deployed service, it creates a target entry with 0% traffic portion
+ - For updating tags of existing traffic targets, use `kn service untag-target` followed by `kn service tag-target`, because
+    - The deployed service might have multiple tags referring to a `RevisionName`
+    - The specified `RevisionName:Tag` combination on CLI might request to have different tags for same revision
+
+#### 2. `kn service untag-target [RevisionName:Tag] [RevisionName [--all]] [Tag]`
```

Comment:
Since 'tag' must be unique across all traffic blocks, can't you just make it:
`kn service untag-target TagName` ?

Reply:
Yes, we can implement that as first iteration to require only `TagName`. The subsequent iteration though should support the other options too.